### PR TITLE
OmniscapeImpact updates for SyncroSim v3

### PR DESCRIPTION
**Pull request by d-romualdi**

### CODE REVIEW

File: src/package.xml [hunk 1]
```diff
@@ -1,126 +1,96 @@
...
+    <column name="dissimilarity" displayName="Dissimilarity" dataType="Double" validationType="Decimal" validationCondition="None" />
+  </dataSheet>
+
+  <!--Transformers-->
+
+  <transformer 
+    name="omniscapeImpactTransformer" 
+    displayName="3 - Connectivity Impact Assessment" 
+    isPrimary="True" 
+    isPipelineBased="True"
```
Comment (d-romualdi) [inline src/package.xml]: Should `isPipelineBased` be changed to 'isPipeline'? Or leave as is?

File: src/package.xml [hunk 2]
```diff
@@ -1,126 +1,96 @@
...
-            <group name="Spatial" displayName="Spatial">
-              <item name="outputSpatialOverall" showIf="inPipeline" />
-              <item name="outputSpatialCategory" showIf="inPipeline" />
-            </group>
-            <group name="Tabular" displayName="Tabular">
-              <item name="outputTabularDifferences" showIf="inPipeline" />
-              <item name="outputTabularChange" showIf="inPipeline" />
-              <item name="outputTabularJaccard" showIf="inPipeline" />
-            </group>
+<package name="omniscapeImpact" displayName="Calculates changes in connectivity categories from a baseline" isAddOn="True" extendsPackage="omniscape" url="https://apexrms.github.io/omniscape/tutorials/omniscapeImpact" version="2.0.0">
```
Comment (katieb1) [inline src/package.xml]: this package element is still syncrosim 2 - reference BurnP3+Prometheus/Cell2Fire to see what it should be

File: src/package.xml [hunk 3]
```diff
@@ -1,126 +1,96 @@
...
+  </dataSheet>
+
+  <dataSheet name="outputTabularJaccard" displayName="Jaccard Dissimilarity">
+    <column name="movementTypesID" displayName="Connectivity category" dataType="Integer" validationType="Datasheet" formula1="omniscape_movementTypes" allowDbNull="False" />
+    <column name="dissimilarity" displayName="Dissimilarity" dataType="Double" validationType="Decimal" validationCondition="None" />
+  </dataSheet>
+
+  <!--Transformers-->
+
+  <transformer
```
Comment (katieb1) [inline src/package.xml:43]: Some of these transformer tags are outdated - compare with BurnP3+Prometheus/Cell2Fire to see what was removed for those packages

File: src/package.xml [hunk 4]
```diff
@@ -1,126 +1,96 @@
...
+      <dataSheet name="outputTabularChange" type="Output" />
+      <dataSheet name="outputSpatialOverall" type="Output" />
+      <dataSheet name="differenceScenarios" type="Input" />
+      <dataSheet name="outputSpatialCategory" type="Output" />
+  </transformer>
+
+  <!--Layouts-->
+
+  <layout type="Scenario" insertAfter="omniscape_Advanced">
+    <group name="Add0on" displayName="Add-on">
```
Comment (katieb1) [inline src/package.xml]: Think about the names of the various tabs and whether they still make sense

File: src/package.xml [hunk 5]
```diff
@@ -1,126 +1,96 @@
...
-        <item name="outputSpatialCategory_perCategoryRaster" displayName="Per category difference" dataSheet="outputSpatialCategory" column="perCategoryRaster" filter="movementTypesID" />
-      </group>
-    </layout>
-  </layouts>
+    </group>
+  </layout>
+
+  <layout type="Chart">
+    <group name="ImpactAssessment" displayName="Impact Assessment">
+      <item name="outputTabularDifferences_amountAreaDifference" displayName="Area difference" dataSheet="outputTabularDifferences" column="amountAreaDifference" filter="movementTypesID" prefixFolderName="False" />
```
Comment (katieb1) [inline src/package.xml]: I don't know if these chart names will work anymore with the "_" - test and fix accordingly

**Review by katieb1 — CHANGES_REQUESTED** [review]

**Comment by katieb1** [comment]
Before requesting another review, I would test loading the new package.xml into SyncroSim Studio to:
(1) make sure it works and installs properly; and
(2) all the tabs/nodes/datasheets look good and make sense

File: src/omniscapeImpactTransformer.py [hunk 6]
```diff
@@ -14,8 +14,8 @@
 packagesInstalled = mySession.packages()
 omniscapeVersion = packagesInstalled.Version[packagesInstalled.Name == "omniscape"]
 
-if pd.unique(omniscapeVersion)[0] != "1.1.2":
-    sys.exit("The omniscapeImpact add-on package version 1.0.2 requires the omniscape base package version 1.1.2.")
+if pd.unique(omniscapeVersion)[0] != "2.1.1":
```
Comment (katieb1) [inline src/omniscapeImpactTransformer.py]: Hm, I think we can delete this bit of code. SyncroSim will already check to make sure the specified version of omniscape is loaded from the `builtFor` tag in the package.xml

File: src/package.xml [hunk 7]
```diff
@@ -1,126 +1,130 @@
...
+      validationType="Decimal" validationCondition="None" />
+  </dataSheet>
+
+  <!--Transformers-->
+
+  <transformer
+    name="omniscapeImpactTransformer"
+    displayName="3 - Connectivity Impact Assessment"
+    transformerSpecificOutput="False"
+    isRunnable="True"
```
Comment (katieb1) [inline src/package.xml]: This element no longer exists and can be removed

File: src/package.xml [hunk 8]
```diff
@@ -1,126 +1,130 @@
...
+    <dataSheet name="outputSpatialOverall" type="Output" />
+    <dataSheet name="differenceScenarios" type="Input" />
+    <dataSheet name="outputSpatialCategory" type="Output" />
+  </transformer>
+
+  <!--Layouts-->
+
+  <layout type="Scenario" insertAfter="omniscape_Advanced">
+    <group name="Impact0Assessment" displayName="Impact Assessment">
+      <item name="differenceScenarios" showIf="inPipeline" />
```
Comment (katieb1) [inline src/package.xml]: `showIf="inPipeline"` no longer exists and can be removed throughout

File: src/package.xml [hunk 9]
```diff
@@ -1,126 +1,130 @@
...
+    <dataSheet name="outputTabularDifferences" type="Output" />
+    <dataSheet name="outputTabularChange" type="Output" />
+    <dataSheet name="outputSpatialOverall" type="Output" />
+    <dataSheet name="differenceScenarios" type="Input" />
+    <dataSheet name="outputSpatialCategory" type="Output" />
+  </transformer>
+
+  <!--Layouts-->
+
+  <layout type="Scenario" insertAfter="omniscape_Advanced">
```
Comment (katieb1) [inline src/package.xml]: I don't think the `insertAfter` exists anymore. This used to tell SyncroSim where to put the datasheets in the tree, but now each package has it's own tab, so this no longer does anything.

**Review by katieb1 — CHANGES_REQUESTED** [review]
Really nice work! Just a few things that can be removed from the script and package.xml. You may also want to arrange a quick call with Carina to show her how the package shows up in the UI to make sure she's happy with the layout.

File: src/package.xml (hunk 7 above)
Comment (d-romualdi) [reply src/package.xml]: Just noticed that the transformerSpecificOutput element is in the omniscape 2.1.1 
package.xml

**Pull request merged**